### PR TITLE
Fix remaining types of code blocks

### PR DESCRIPTION
**Pull request by ST-DDT**
Fixes the remaining code blocks that use the wrong type definition for their content and also document which type should be used for which content.

### CODE REVIEW

**Comment by Spongy** [comment]
A preview for this pull request is available at https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/21d2ee9/.

Here are some links to the pages that were modified:

- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/21d2ee9/about/tos.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/21d2ee9/contributing/howtogit.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/21d2ee9/contributing/implementation/datamanipulator.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/21d2ee9/contributing/implementation/pr.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/21d2ee9/contributing/spongedocs.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/21d2ee9/plugin/configuration/nodes.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/21d2ee9/plugin/practices/bad.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/21d2ee9/plugin/wgen/modifiers.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/21d2ee9/preparing/git.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/21d2ee9/server/getting-started/configuration/hocon.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/21d2ee9/server/getting-started/configuration/sponge-conf.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/21d2ee9/server/getting-started/launch-script.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/21d2ee9/server/management/exploit-patches.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/21d2ee9/server/management/performance-tweaks.html

_Since the preview frequently changes, please link to [this comment](https://github.com/SpongePowered/SpongeDocs/pull/717#issuecomment-403621037), not to the direct url to the preview._

File: source/contributing/implementation/pr.rst [hunk 1]
```diff
@@ -51,7 +51,7 @@ A few things that can be taken from this:
 
 * Links to any implementation PRs in clear view at the top of the PR, this can be achieved with GitHub Markdown
 
-.. code-block:: none
+.. code-block:: text
```
Comment (ST-DDT) [inline source/contributing/implementation/pr.rst]: Not sure on this one: Should we use `md` instead?

File: source/contributing/spongedocs.rst [hunk 2]
```diff
@@ -52,6 +52,17 @@ writing Sponge Documentation. This list may get added to (or bent out of shape)
      Contributors are discouraged from commenting in code blocks wherever possible. Simple place-holder text may be
      necessary in some examples. Ideally, code block examples will be short, and followed by an explanation for each
      example in the body text. Of course, there may be some concepts that cannot be illustrated with a short example.
+  ii. Use the following code block types for particular code blocks:
+
+     * Groovy/Gradle-Files -> ``groovy``
+     * Hocon-Config -> ``guess`` (hocon highlighting is not yet supported)
```
Comment (ST-DDT) [inline source/contributing/spongedocs.rst:58]: Not sure on this one: `guess` will hardly find the right one. `groovy` yields good results but is logically wrong. `js`/`json` fail on comments.

File: source/contributing/spongedocs.rst [hunk 3]
```diff
@@ -52,6 +52,17 @@ writing Sponge Documentation. This list may get added to (or bent out of shape)
...
+     * In-Game/Server Console Command -> ``none``
```
Comment (ST-DDT) [inline source/contributing/spongedocs.rst:59]: Not sure on this one: Any suggestions?

File: source/contributing/spongedocs.rst [hunk 4]
```diff
@@ -52,6 +52,17 @@ writing Sponge Documentation. This list may get added to (or bent out of shape)
...
+     * Java-Code -> ``java``
+     * Json-Config -> ``json``
+     * Linux-Terminal -> ``bash``
+     * Logs -> ``none``
```
Comment (ST-DDT) [inline source/contributing/spongedocs.rst:63]: Not sure on this one: Maybe `text`?

File: source/contributing/spongedocs.rst (hunk 2 above)
Comment (ST-DDT) [reply source/contributing/spongedocs.rst:58]: Ref: https://github.com/SpongePowered/SpongeDocs/issues/713 Add code highlighting support for hocon examples

File: source/contributing/implementation/pr.rst (hunk 1 above)
Comment (12awsomeman34) [reply source/contributing/implementation/pr.rst]: md but you need to change the line below to add spaces around the  | for it to format correctly
md_markdown

File: source/contributing/spongedocs.rst (hunk 4 above)
Comment (12awsomeman34) [reply source/contributing/spongedocs.rst:63]: Text doesn't highlight anything, so its a matter of preference in the rst file.

**Review by parlough — APPROVED** [review]

**Comment by Inscrutable** [comment]
If we find better alternatives for the other options we can shoe-horn them in later. This looks ready to merge, unless anyone has further input?

**Review by 12awsomeman34 — APPROVED** [review]
If a hocon lexer is ever made then the other two (logs and commands) wouldn't be too difficult to make either. Regardless, it would be somewhat overkill to add highlighting for those and they can be left as `none` for the time being.

**Pull request merged**